web-service/helpers.py

```diff
@@ -132,42 +130,48 @@ def SortByDistanceCmp(a, b):
 
 ################################################################################
 
-def FetchAndStoreUrl(siteInfo, url):
+def GetSiteInfoForUrl(url, rssi=None, txpower=None, force_update=False):
+    siteInfo = models.SiteInformation.get_by_id(url)
```

Comment:
Exactly.  This is by design, most redirects specifically ask to *not* be cached.

We have TODO's to actually read cache-content headers and respect them, but we aren't doing this even for the final destination, which is really where the win will be.